code/modules/mob/mob.dm

```diff
@@ -1345,9 +1345,22 @@ var/list/slot_equipment_priority = list( \
 /mob/proc/update_sight()
 	SEND_SIGNAL(src, COMSIG_MOB_UPDATE_SIGHT)
 	sync_lighting_plane_alpha()
+	sync_nightvision_screen()
 
 /mob/proc/sync_lighting_plane_alpha()
 	if(hud_used)
 		var/obj/screen/plane_master/lighting/L = hud_used.plane_masters["[LIGHTING_PLANE]"]
-		if (L)
-			L.alpha = lighting_alpha
\ No newline at end of file
+		if(L)
+			L.alpha = lighting_alpha
+
+/mob/proc/sync_nightvision_screen()
```

Comment:
This is what is responsible for the stuff to be dynamic.

The "area around your mob that's dimly lit" thing is already in existence with TG's lighting system---this proc here just adds 6 more screens that have a bigger circumference, ergo, any time this proc is called, it'll make the vision sync with the mob's `see_in_dark`, thus mimicing old behavior.

Reply:
I agree, however it is more static than the alpha animation since this has overlays that do not change and are not calculated by anything more than a switch statement

Reply:
Ah yeah, this is done in a sort of part1 part2 deal. This being the mainstay to mimic old behaviour while truly new behaviour, part2 - the adjustment to dark with the animation of alpha on the overlay is done in `handle_darksight()`